doc/mgmt/SONiC_PIM_SM_HLD.md

```diff
@@ -0,0 +1,1036 @@
+| IGMP      | Internet Group Management Protocol                                |
+| LHR       | Last Hop Router (Router directly connected to the Host/Receiver   |
+| MLD       | Multicast Listener Discovery Protocol                                      |
+| RP        | Rendezvous Point                                                  |
+| SPT       | Shortest Path Tree                                                |
+| RPF       | Reverse Path Forwarding                                           | 
+| (S,G)     | (Source address, Group address)                                   |
+| NPU       | Network processor (NPU) is analogous to central processing unit in a computer or similar device       |
+| OIF/OIL   | Outgoing Interface List                                           |
+| PIM-SM    | PIM - Sparse Mode                                                 |
+| PIM-ASM   | PIM - Any Source Multicast                                        |
+| VRF       | Virtual Router Forwarding                                         |
+
+
+# 1 Feature Overview
+
+PIM Any Source Multicast(ASM) control plane support is already available in Sonic FRR. But, the interaction of FRR with the other modules for programming (*,G) routes in Hardware, lifting unknown multicast and known multicast routes are not available. These changes are defined in this document. It also covers the configuration support for PIM-ASM through SONIC-CLI.
+
+
+## 1.1 Requirements
+### 1.1.1 Functional Requirement
+1. Support trapping of unknown ipv4 multicast data to CPU through a dedicated CoPP queue.
+2. Support trapping of known ipv4 multicast data to CPU through a dedicated CoPP queue.
+3. Support programming of (*,G) routes into hardware using SAI APIS.
+4. Support programming of register interface as CPU port in (*,G) and (S,G) multicast routes
+5. Support IPv4 multicast forwarding on Port based, VLAN based and LAG based routing interfaces.
+6. Provide support for VRF in PIM ASM.
+7. Support the liveliness tracking of (S,G) route through HW timers and programming the HW timer value using SAI APIS.
+8. Support to stop multicast data forwarding in Kernel to avoid duplication of traffic.
+9. Support for static configuration of RP 
+
+### 1.1.2 Configuration and Management Requirements
+1. Support to configure and display PIM-SM (IPv4) informations using KLISH CLI commands.
+2. Support to configure and GET PIM-SM informations using REST and gNMI.
+3. Support for GNMI subscriber
+4. Support save and reload of PIM-SM configurations.
+5. Support to display PIM SM details in "show running-config" and also add "show running-config pim"
+
+### 1.1.3 Scalability Requirements
+The following are the scalability requirements for this feature:
+1. Maximum number of PIM neighbors supported is 64.[ User can enable 64 pim interfaces in a single VRF (or) can split across 64 vrfs]   
+2. Maximum number of multicast route entries supported is 8K, this is inclusive of snooping and pim routes.
```

Comment:
There is no restriction on number of static RPs, since we claim only 32 interfaces can be PIM enabled we have to restrict the static RP also to that number. As we are yet to bench mark as of now we will retain it as 31.  There is no restriction in the ACL mapped to static RP